client/core/core.go

```diff
@@ -3565,6 +3567,54 @@ func (c *Core) Login(pw []byte) (*LoginResult, error) {
 	}
 	defer crypter.Close()
 
+	loginResult := func() *LoginResult {
+		// NOTE: initializeDEXConnections will do authDEX and a 'connect'
+		// request only if NOT already dc.acct.authed().
+		dexStats := c.initializeDEXConnections(crypter)
+		notes, err := c.db.NotificationsN(100)
+		if err != nil {
+			c.log.Errorf("Login -> NotificationsN error: %v", err)
+		}
+
+		return &LoginResult{
+			Notifications: notes,
+			DEXes:         dexStats,
+		}
+	}
+
+	// Connecting and loading wallets might take time, but we don't want
+	// concurrent wallet connect attempts running, especially for native wallets
+	// that use the same file system resources. NOTE: The following channel
+	// mechanism similar to the following, and it does suggest some design
+	// issues, described below.
+	//
+	// !c.loginMtx.TryLock() {
+	//  c.loginMtx.Lock()
+	//  defer c.loginMtx.Lock()
+	//  return loginResult(), err
+	// }
+	defer func() { <-c.loginSlot }() // remove the peg from the slot when done
+	select {
+	case c.loginSlot <- struct{}{}: // full login
+	default:
+		c.log.Warnf("Login already running! Waiting for it to complete.")
+		// Just wait and take the short path.
+		c.loginSlot <- struct{}{}
+		return loginResult(), err
```

Comment:
This contraption has not been giving any trouble on master since it was merged on Aug 22, so I think we can use this on the release branch as well to prevent concurrent logins from doing bad things.